### PR TITLE
Add upgrade rule to check for mesos executor and flag to change it.

### LINKED ISSUE #11043: Create MesosExecutorRemovedRule to ease upgrade to Airflow 2.0
This issue is part of #8765

## Rule
Create `MesosExecutorRemovedRule` which corresponds to
> Removal of Mesos Executor

entry in UPDATING.md. This rule should allow users to check if their current configuration needs any adjusting
before migration to Airflow 2.0.

## How to guide
To implement a new rule, create a class that inherits from `airflow.upgrade.rules.base_rule.BaseRule`.
It will be auto-registered and used by `airflow upgrade-check` command. The custom rule class has to have `title`,
`description` properties and should implement `check` method which returns a list of error messages in case of
incompatibility.

For example:
https://github.com/apache/airflow/blob/ea36166961ca35fc51ddc262ec245590c3e236fb/airflow/upgrade/rules/conn_type_is_not_nullable.py#L25-L42

**Remember** to open the PR against `v1-10-test` branch.

### PR DESCRIPTION
**Pull request by RaviTezu**
closes: #11043

### CODE REVIEW

File: airflow/upgrade/rules/mesos_executor_removed.py
Comment (mik-laj) [inline airflow/upgrade/rules/mesos_executor_removed.py]: The existence of this section in the configuration file does not prevent migration to Airflow 2.0.  I think we should check that the `executor` option in the `core` section is not set to `MesosExecutor` and warn the user that he needs to migrate to another supported alternative such as CeleryExecutor.

Comment (RaviTezu) [reply airflow/upgrade/rules/mesos_executor_removed.py]: Thanks for the reply @mik-laj I will make the changes.

**Comment by RaviTezu** [comment]
@mik-laj Could you please take a look now?

**Review by turbaszek — APPROVED** [review]

File: airflow/upgrade/rules/mesos_executor_removed.py
Comment (turbaszek) [inline airflow/upgrade/rules/mesos_executor_removed.py]: Please remove the leading new line and indent

Comment (RaviTezu) [reply airflow/upgrade/rules/mesos_executor_removed.py]: Thanks for the reply @turbaszek. Pushed the changes. Please take a look. 
@mik-laj FYI

Comment (RaviTezu) [reply airflow/upgrade/rules/mesos_executor_removed.py]: @turbaszek @mik-laj Can we go ahead and merge this PR? Let me know if you need anything else on this. Thanks.

**Comment by turbaszek** [comment]
@mik-laj can you please take a look at CI? This look like a problem with building docs - but not related to the PR?

**Comment by RaviTezu** [comment]
@mik-laj Could you please check this PR? Thanks.

**Comment by turbaszek** [comment]
@RaviTezu  would you mind rebasing and changing the target base due to https://github.com/apache/airflow/pull/11719#issuecomment-714258732

**Comment by RaviTezu** [comment]
@turbaszek Pulled (with --rebase) commits from `v1-10-test` into my branch and then changed the base to `v1-10-stable`. There was a merge conflict and I resolved it.

**Comment by turbaszek** [comment]
@RaviTezu sorry for not being clear, you should not only change the target for `v1-10-stable` but also rebase on this branch. Sorry for inconvenience

**Comment by RaviTezu** [comment]
@turbaszek No worries. Rebased and forced pushed.

**Comment by RaviTezu** [comment]
@turbaszek If this looks good to you. Could you please merge the PR? Thanks.

**Comment by turbaszek** [comment]
@RaviTezu I cleaned up the PR, once the CI is green I will merge it

**Comment by RaviTezu** [comment]
Thank you @turbaszek

**Comment by RaviTezu** [comment]
@turbaszek I looked at the failing checks and they seem to be either the CI/CD system related issues or some other test seems to be failing. 

**build errors:**
```
{"success":false,"error":400,"message":"Trouble uploading file"}###########################################################################################
                   EXITING /opt/airflow/scripts/ci/in_container/run_ci_tests.sh WITH STATUS CODE 1
```

**Tests failing:**
```
test_scheduler_run_duration
tests/executors/test_dask_executor.py::DaskExecutorTest::test_backfill_integration
```

**Comment by potiuk** [comment]
Linked to #11391

**Comment by kaxil** [comment]
Can you please rebase your PR on latest Master since we have applied [Black](https://github.com/apache/airflow/commit/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68) and [PyUpgrade](https://github.com/apache/airflow/commit/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5) on Master.

It will help if your squash your commits into single commit first so that there are less conflicts.

**Pull request merged**
